### PR TITLE
1154 - my deals bug

### PR DESCRIPTION
**Pull request by IonelLupu**
## What was done
Fixed "my deals" button when you have "Partnered deals" tab selected.

First of all, I fixed a bug that showed up when you change the Metamask address. Aurelia, for some reason, cannot handle async router lifecycle hooks. The error was hard to debug but, at the end, I had to remove the `async` keyword from the `attached` lifecycle hook.

This fixed the issue, but in the mean time, I found that the "Open proposal" and "partnered deals" tabs were not displayed if the selected one didn't have any deals. So I fixed that as well. It now shows the other tab if the selected one doesn't have any deals

#### After

### CODE REVIEW

File: src/deals/list/deals.ts [hunk 1]
```diff
@@ -174,14 +174,10 @@ export class Deals {
     this.showMine = !this.showMine;
     if (this.showMine) {
       //if showing only "my deals" check to see which tab to display by default if there are no deals in either tab
-      const openDeals = this.isTabVisible(0, this.showMine, this.ethereumService.defaultAccountAddress);
-      const partneredDeals = this.isTabVisible(1, this.showMine, this.ethereumService.defaultAccountAddress);
-      if (openDeals) {
+      const hasPartneredDeals = this.isTabVisible(1, this.showMine, this.ethereumService.defaultAccountAddress);
+      if (!hasPartneredDeals) {
         this.cardIndex = 0;
       }
```
Comment (hiaux0) [inline src/deals/list/deals.ts:180]: Can you give a little background of why the "openDeals" case is not needed anymore? (because default?)

**Review by hiaux0 — COMMENTED** [review]
fncality -> fixed

File: src/deals/list/deals.ts (hunk 1 above)
Comment (IonelLupu) [reply src/deals/list/deals.ts]: You are right. I tested more and it seems I have to add the other condition back. I also fixed another old one we had when the account was changed in metamask. (check the removed `async` for that)

**Review by hiaux0 — CHANGES_REQUESTED** [review]
Change request: Looks like you only are filtering for proposal lead?
"a deal is “mine” if I am either the Proposal Lead or a Representative" (https://www.notion.so/primedao/All-Deals-My-Deals-7493545968cb4c0f965e2ec249716310#21393bab719144e5bfa1e5d41931d323)

1. connect as 0xBf3a5599f2f6CE89862d640a248e31F30B7ddF29
2.1 refac/au2 branch -> shows also me as a rep
2.2 your branch -> only prop lead?

**Comment by IonelLupu** [comment]
> Change request: Looks like you only are filtering for proposal lead? "a deal is “mine” if I am either the Proposal Lead or a Representative" (https://www.notion.so/primedao/All-Deals-My-Deals-7493545968cb4c0f965e2ec249716310#21393bab719144e5bfa1e5d41931d323)
> 
> 1. connect as 0xBf3a5599f2f6CE89862d640a248e31F30B7ddF29
>    2.1 refac/au2 branch -> shows also me as a rep
>    2.2 your branch -> only prop lead?

Good find. Fixed

**Review by hiaux0 — APPROVED** [review]

**Pull request merged**